akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -495,9 +495,7 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
         changeState(recovering(recoveryBehavior, previousRecieveTimeout))
         journal ! ReplayMessages(lastSequenceNr + 1L, toSnr, replayMax, persistenceId, self)
       case ReceiveTimeout ⇒
-        try onRecoveryFailure(
-          new RuntimeException(s"Recovery timed out, didn't get snapshot within timeout"),
-          event = None) finally context.stop(self)
+        try onRecoveryFailure(new RecoveryTimedOut, event = None) finally context.stop(self)
```

Comment:
add a message to the exception, include the `lastSequenceNr` and the timeout value in the message (yeah, almost like the previous version)